File: Linked List/Remove Nth Node From End of List/RemoveNthNodeFromEnd.cpp
class Solution {
public:
    ListNode* removeNthFromEnd(ListNode* head, int n) {
        int counter = 0;
        ListNode* current = head;
        while (current != nullptr) {
            current = current->next;
            counter++;
        }
        current = head;
        if (counter == n) {
            head = head->next;
            delete current;
            return head;
        }
        for (int i = 0; i < counter - n - 1; i++) {
            current = current->next;
        }
        ListNode* temp = current->next;
        current->next = current->next->next;
        delete temp;
        return head;
    }
};

File: Linked List/Remove Nth Node From End of List/RemoveNthNodeFromEnd.cs
public class Solution {
    public ListNode RemoveNthFromEnd(ListNode head, int n) {
        int counter = 0;
        ListNode current = head;
        while (current != null) {
            current = current.next;
            counter++;
        }
        current = head;
        if (counter == n) {
            head = head.next;
            return head;
        }
        for (int i = 0; i < counter - n - 1; i++) {
            current = current.next;
        }
        ListNode temp = current.next;
        current.next = current.next.next;
        return head;
    }
}